<launch>
  <!-- RealSense D456 Ä«¸Þ¶ó ¼³Á¤ -->
  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="enable_depth" value="false"/>
    <arg name="color_width" value="1280"/>
    <arg name="color_height" value="720"/>
  </include>

  <!-- Velodyne VLP-32 ¶óÀÌ´Ù ¼³Á¤ -->
  <include file="$(find velodyne_pointcloud)/launch/VLP32_points.launch">
    <arg name="device_ip" value="192.168.1.201"/>
    <arg name="port" value="2368"/>
    <arg name="frame_id" value="velodyne"/>
  </include>

  <!-- Â÷¼± ÀÎ½Ä ³ëµå -->
  <node pkg="lane_follower_pkg" type="lane_detection.py" name="lane_detector" output="screen">
    <param name="camera_topic" value="/camera/color/image_raw"/>
  </node>

  <!-- ÁÖÇà Á¦¾î ³ëµå -->
  <node pkg="lane_follower_pkg" type="control_node.py" name="controller" output="screen">
    <param name="max_speed" value="0.5"/>
    <param name="steering_gain" value="1.2"/>
  </node>

  <!-- Àå¾Ö¹° È¸ÇÇ ³ëµå (¿É¼Ç) -->
  <node pkg="lane_follower_pkg" type="obstacle_avoidance.py" name="obstacle_avoider" output="screen">
    <param name="lidar_topic" value="/velodyne_points"/>
  </node>

  <!-- RViz ½ÇÇà -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lane_follower_pkg)/rviz/lane_follower.rviz"/>
</launch>
